src/dotnet/ExecutablePackageObtainer/ExecutablePackageObtainer.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.ExecutablePackageObtainer
+{
+    public class ExecutablePackageObtainer
+    {
+        private readonly Lazy<string> _bundledTargetFrameworkMoniker;
+        private readonly Func<FilePath> _getTempProjectPath;
+        private readonly ICanAddPackageToProjectFile _packageToProjectFileAdder;
+        private readonly ICanRestoreProject _projectRestorer;
+        private readonly DirectoryPath _toolsPath;
+
+        public ExecutablePackageObtainer(
+            DirectoryPath toolsPath,
+            Func<FilePath> getTempProjectPath,
+            Lazy<string> bundledTargetFrameworkMoniker,
+            ICanAddPackageToProjectFile packageToProjectFileAdder,
+            ICanRestoreProject projectRestorer
+        )
+        {
+            _getTempProjectPath = getTempProjectPath;
+            _bundledTargetFrameworkMoniker = bundledTargetFrameworkMoniker;
+            _projectRestorer = projectRestorer ?? throw new ArgumentNullException(nameof(projectRestorer));
+            _packageToProjectFileAdder = packageToProjectFileAdder ??
+                                         throw new ArgumentNullException(nameof(packageToProjectFileAdder));
+            _toolsPath = toolsPath;
+        }
+
+        public ToolConfigurationAndExecutableDirectory ObtainAndReturnExecutablePath(
+            string packageId,
+            string packageVersion = null,
+            FilePath? nugetconfig = null,
+            string targetframework = null)
+        {
+            if (packageId == null) throw new ArgumentNullException(nameof(packageId));
```

Comment:
The formatting for this conditional should probably have braces around the statements, even for a single line.

Reply:
Fixed